### PR TITLE
feat(channels): auto-refresh LINE access tokens before they expire

### PR DESCRIPTION
**Pull request by devxoul**
## Background

A live agent stopped responding on LINE after a container restart, with the cryptic startup error `adapter "line" failed to start: undefined is not an object (evaluating 'res.data.e.code')`. Root cause was twofold:

1. **Token expiry, never renewed.** The LINE adapter (`src/channels/adapters/line.ts`) logs in once at start with the stored `auth_token` — a ~7-day JWT — and keeps it for the process lifetime. There was no renewal path at all. Once the JWT expired, every `start()` failed and the channel stayed down until a manual QR/email re-auth.

2. **The SDK's refresh is reactive-only.** agent-messenger's vendored linejs has refresh machinery (`AuthService.tryRefreshToken()` driven off `update:authtoken`), but it only fires when a *live request* gets a `MUST_REFRESH_V3_TOKEN` response. The wrapper's `login()` starts from the stale JWT, so a fully-expired token at boot fails server-side before the refresh path can ever run. (The opaque `TypeError` is a separate SDK null-guard bug, handled upstream.)

The account already has everything renewal needs: the LINE refresh token lives in the SDK's per-account `FileStorage` (bind-mounted, survives restarts, reachable in-container), is valid ~365 days, and rotates on use. So unlike Webex/KakaoTalk — which need the host keystore to decrypt a stored password and therefore renew host-side (#1001) — LINE can refresh **in-container without a restart**: the SDK mutates the live client's token in place and emits an event. This PR builds the consumer that was missing.

## Summary

Drive refresh proactively from the adapter, in three small pieces:

- **`line-token.ts`** — decode the `auth_token` JWT `exp` and compute when it enters a 24h refresh skew window, clamped to a daily re-check. Pure, unit-tested.
- **Adapter wiring** — at startup and on a self-rescheduling timer, call the SDK's `ensureFreshAuthToken()`; also subscribe to `onTokenUpdate` for tokens the SDK rotates on its own (e.g. reactive `MUST_REFRESH`). Every refreshed access token is mirrored back into `secrets.json` so the next restart boots from a live token.

**Why X not Y:**

- **Why in-container, not a hostd cron like Webex/Kakao?** Those are host-side *only* because renewal needs the keystore to decrypt a password, and the live client holds a stale token in a closure → must restart. Neither applies to LINE: no password/keystore (refresh token is plaintext in a host-and-container-readable file), and the SDK updates the live client in place, so no restart is needed. A host cron would add restart churn and a host/container double-refresh race against a *rotating* token.
- **Why are `ensureFreshAuthToken`/`onTokenUpdate` optional on the client type?** Same structural duck-typing the file already uses for `LineClient`. An SDK build that predates the refresh surface degrades to "no proactive refresh" instead of crashing — no hard version pin.
- **Why never copy the refresh token into `secrets.json`?** It rotates on use; a stale copy is worse than none. The SDK `FileStorage` stays the single source of truth for refresh state; `secrets.json` only holds the current *access* token snapshot needed to bootstrap.

## What this does NOT do

- Does not touch agent-messenger. It depends on the SDK exposing `ensureFreshAuthToken()` + `onTokenUpdate` and fixing the expired-token `login()` fallback / `res.data.e.code` null-guard — those land in a separate upstream PR. This adapter feature-detects both and is a safe no-op until they ship.
- Does not add a hostd LINE renewal cron (see Summary for why).
- Does not change the LINE channel/credential schema.
- Does not handle a fully-expired *refresh* token (~365d) — that still requires interactive QR/email re-auth, a LINE platform constraint.

## Review notes

- **Load-bearing invariant:** the rotating refresh token is never written to `secrets.json` (`persistAuthToken` only ever sets `auth_token`). Violating this strands a dead token across restarts.
- **Non-fatal by design:** all refresh failures are caught and logged; a refresh error can never take the channel down at start. Covered by the "refresh failure at startup is non-fatal" and "SDK version skew" tests.
- **Timer:** `unref()`'d and self-reschedules off the freshest token's `exp`; cleared in `stop()`. The daily max-delay clamp means the timer re-evaluates at most once a day rather than sleeping for the full multi-day gap.
- Tests: `line-token.test.ts` (decode/window/clamp edge cases) and four new adapter cases (startup refresh persists; `onTokenUpdate` persists; failure non-fatal; missing-surface no-op). Full suite green locally: typecheck, lint (0 errors), format:check, 9639 pass / 0 fail.

### CODE REVIEW

File: src/channels/adapters/line.ts [hunk 1]
```diff
@@ -199,6 +210,50 @@ export function createLineAdapter(options: LineAdapterOptions): LineAdapter {
 
   const outboundCallback = createOutboundCallback({ client, logger, formatChannelTag })
 
+  // The SDK's AuthService updates the LIVE client's token in-place and emits this
+  // event; we mirror it into secrets.json so the next container start boots from a
+  // fresh token instead of a dead one. The refresh token (which rotates on use)
+  // stays the SDK FileStorage's responsibility and is never copied here.
+  const persistAuthToken = async (authToken: string): Promise<void> => {
+    if (authToken === currentAuthToken) return
+    currentAuthToken = authToken
```
[the remark this answers is not in the document]
Comment (devxoul) [reply src/channels/adapters/line.ts:223]: Fixed. Split the conflated state: `currentAuthToken` now only drives refresh scheduling and always updates, while `lastPersistedToken` is the dedupe marker and advances only after `setAccount` resolves. A failed write therefore leaves the dedupe marker untouched, so a later `onTokenUpdate`/`ensureFreshAuthToken` with the same live token retries the write. Also seeded `lastPersistedToken` from the stored token at startup to avoid a redundant rewrite of a still-valid token. Added the requested regression test (`setAccount` throws once, same token retries and writes on the next callback). Squashed into the original commit.
